pkg/grpc/server.go

```diff
@@ -313,22 +330,28 @@ func (s *server) startWorkloadCertRotation() {
 	s.logger.Infof("starting workload cert expiry watcher. current cert expires on: %s", s.signedCert.Expiry.String())
 
 	ticker := time.NewTicker(certWatchInterval)
```

Comment:
Unrelated from this PR but why are we checking if the cert is expired on a regular interval? We know when the cert expires, so we should simply wait until that needs to be renewed, no? That avoids getting locks for no reason.

Reply:
Yeah it was just an unrelated thing. Good that it's getting ripped anyways